crates/pixi_pty/src/unix/pty_session.rs

```diff
@@ -80,6 +81,9 @@ impl PtySession {
     /// forward all input from stdin to the process and all output from the process to stdout.
     /// This will block until the process exits.
     pub fn interact(&mut self, wait_until: Option<&str>) -> io::Result<Option<i32>> {
+        let pattern_timeout = 4;
```

Comment:
Can you make this a duration? Then we dont have to guess about whether this is seconds or ms.